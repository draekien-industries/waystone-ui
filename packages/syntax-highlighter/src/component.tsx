import { CodeLine } from './components';
import type {
  SyntaxHighlightStyle as StyleName,
  SupportedLanguage,
} from './lib';
import { SYNTAX_HIGHLIGHT_STYLESHEETS } from './lib';
import { createCodeTree, flattenCodeTree } from './utils';

export type SyntaxHighlighterProps = {
  language?: SupportedLanguage;
  styleName?: StyleName;
  wrapLines?: boolean;
  children: string;
};

export const SyntaxHighlighter = ({
  children,
  wrapLines,
  language,
  styleName = 'defaultStyle',
}: SyntaxHighlighterProps) => {
  const style = SYNTAX_HIGHLIGHT_STYLESHEETS[styleName];
  const { children: tree } = createCodeTree({ code: children, language });
  const codeTree = flattenCodeTree({ tree });

  return (
    <pre
      sx={{
        ...style.hljs,
        paddingX: 'medium',
        fontFamily: 'inherit',
        fontWeight: 'inherit',
        lineHeight: 'inherit',
        paddingTop: '2rem',
        paddingBottom: '2.25rem',
      }}
    >
      <code
        className={language ? `language-${language}` : undefined}
        sx={{ whiteSpace: wrapLines ? 'pre-wrap' : 'pre' }}
      >
        {codeTree.map((node, index) => (
          <CodeLine stylesheet={style} key={`code-segment-${index.toString()}`}>
            {node}
          </CodeLine>
        ))}
      </code>
    </pre>
  );
};

export default SyntaxHighlighter;
