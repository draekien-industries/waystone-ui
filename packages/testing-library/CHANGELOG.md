# @waystone/testing-library

## 9.2.0

### Minor Changes

- 21db598: chore: bump deps and add documentation

### Patch Changes

- Updated dependencies [21db598]
  - @waystone/theme@9.3.0

## 9.1.6

### Patch Changes

- 41664e2: fix: use type imports to improve bundling
- Updated dependencies [41664e2]
  - @waystone/theme@9.2.5

## 9.1.5

### Patch Changes

- 856ba2b: fix: missing dependencies when installing package
- Updated dependencies [856ba2b]
  - @waystone/theme@9.2.4

## 9.1.4

### Patch Changes

- 5b726aa: fix build and dependencies
- Updated dependencies [5b726aa]
  - @waystone/theme@9.2.3

## 9.1.3

### Patch Changes

- 7086019: Update dependencies
- Updated dependencies [7086019]
  - @waystone/theme@9.2.2

## 9.1.2

### Patch Changes

- @waystone/theme@9.2.1

## 9.1.1

### Patch Changes

- Updated dependencies [8f75dd6]
  - @waystone/theme@9.2.0

## 9.1.0

### Minor Changes

- 5c43357: Switch to pnpm and implement minimal syntax highlighter

### Patch Changes

- Updated dependencies [5c43357]
  - @waystone/theme@9.1.0

## 9.0.0

### Major Changes

- 4a19c42: Update package structure

  Created the `@waystone/ui` package for easier installation and created the `@waystone/testing-library`
  package for a preconfigured testing-library render function.

  Uninstall your `@waystone/*` packages, and run the command `yarn add @waystone/ui`
  to update to the latest version. You will need to update the imports to point
  to the new source.

### Patch Changes

- Updated dependencies [4a19c42]
  - @waystone/theme@9.0.0
