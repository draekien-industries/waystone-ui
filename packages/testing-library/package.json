{
  "name": "@waystone/testing-library",
  "version": "10.0.2",
  "private": true,
  "main": "./src/index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/draekien-industries/waystone-ui"
  },
  "description": "The react component library used by draekien-industries",
  "author": "draekien-industries",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "src/**"
  ],
  "scripts": {
    "lint": "eslint --ext .tsx,.ts,.jsx,.js ./src/ --fix",
    "lint:release": "eslint --ext .tsx,.ts,.jsx,.js ./src/"
  },
  "devDependencies": {
    "@waystone/tsconfig": "workspace:*",
    "eslint-config-waystone": "workspace:*",
    "eslint-plugin-waystone": "workspace:*",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "dependencies": {
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.2.0",
    "@testing-library/user-event": "^14.6.0",
    "@theme-ui/color": "^0.17.1",
    "@theme-ui/css": "^0.17.1",
    "@waystone/theme-provider": "workspace:*",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "theme-ui": "^0.17.1"
  },
  "publishConfig": {
    "access": "public"
  }
}
