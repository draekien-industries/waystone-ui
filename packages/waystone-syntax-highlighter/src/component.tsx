import { CodeLine } from './components';
import { useCodeTree } from './hooks';
import { StyleName, SupportedLanguage, styles } from './lib';

export type SyntaxHighlighterProps = {
  language?: SupportedLanguage;
  styleName?: StyleName;
  wrapLines?: boolean;
  children: string;
};

export const SyntaxHighlighter = ({
  children,
  wrapLines,
  language,
  styleName = 'defaultStyle',
}: SyntaxHighlighterProps) => {
  const style = styles[styleName];

  const codeTree = useCodeTree({
    code: children,
    language,
  });

  return (
    <pre
      sx={{
        ...style.hljs,
        paddingX: 'md',
        fontFamily: 'inherit',
        fontWeight: 'inherit',
        lineHeight: 'inherit',
        paddingTop: '2rem',
        paddingBottom: '2.25rem',
      }}
    >
      <code
        className={language ? `language-${language}` : undefined}
        sx={{ whiteSpace: wrapLines ? 'pre-wrap' : 'pre' }}
      >
        {codeTree.map((node, index) => (
          <CodeLine
            node={node}
            stylesheet={style}
            key={`code-segment-${index.toString()}`}
          />
        ))}
      </code>
    </pre>
  );
};

export default SyntaxHighlighter;
