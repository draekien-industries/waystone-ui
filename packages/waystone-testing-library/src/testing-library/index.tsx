'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
