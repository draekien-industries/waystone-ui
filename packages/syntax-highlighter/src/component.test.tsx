import { render } from '@waystone/testing-library';
import { SyntaxHighlighter } from './component';

describe('Components: SyntaxHighlighter', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <SyntaxHighlighter language="javascript">
        console.log(&quot;Hello World&quot;)
      </SyntaxHighlighter>
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when wrapLines is true', () => {
    const { container } = render(
      <SyntaxHighlighter language="javascript" wrapLines>
        console.log(&quot;Hello World&quot;)
      </SyntaxHighlighter>
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when styleName is not defaultStyle', () => {
    const { container } = render(
      <SyntaxHighlighter language="javascript" styleName="a11yDark">
        console.log(&quot;Hello World&quot;)
      </SyntaxHighlighter>
    );

    expect(container).toMatchSnapshot();
  });
});
