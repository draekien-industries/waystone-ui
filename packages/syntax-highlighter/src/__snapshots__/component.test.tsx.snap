// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: SyntaxHighlighter should match snapshot 1`] = `
<div>
  <pre
    class="css-1qg9ou7-SyntaxHighlighter"
  >
    <code
      class="language-javascript css-dytyo9-SyntaxHighlighter"
    >
      <span
        class="hljs-variable language_ css-g89m8z-Box"
        style="color: rgb(171, 86, 86);"
      >
        <span>
          console
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          .
        </span>
      </span>
      <span
        class="hljs-title function_ css-g89m8z-Box"
        style="color: rgb(136, 0, 0); font-weight: bold;"
      >
        <span>
          log
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          (
        </span>
      </span>
      <span
        class="hljs-string css-g89m8z-Box"
        style="color: rgb(136, 0, 0);"
      >
        <span>
          "Hello World"
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          )
        </span>
      </span>
    </code>
  </pre>
</div>
`;

exports[`Components: SyntaxHighlighter should match snapshot when styleName is not defaultStyle 1`] = `
<div>
  <pre
    class="css-1blygkj-SyntaxHighlighter"
  >
    <code
      class="language-javascript css-dytyo9-SyntaxHighlighter"
    >
      <span
        class="hljs-variable language_ css-g89m8z-Box"
        style="color: rgb(255, 160, 122);"
      >
        <span>
          console
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          .
        </span>
      </span>
      <span
        class="hljs-title function_ css-g89m8z-Box"
        style="color: rgb(0, 224, 224);"
      >
        <span>
          log
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          (
        </span>
      </span>
      <span
        class="hljs-string css-g89m8z-Box"
        style="color: rgb(171, 227, 56);"
      >
        <span>
          "Hello World"
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          )
        </span>
      </span>
    </code>
  </pre>
</div>
`;

exports[`Components: SyntaxHighlighter should match snapshot when wrapLines is true 1`] = `
<div>
  <pre
    class="css-1qg9ou7-SyntaxHighlighter"
  >
    <code
      class="language-javascript css-1fq9u91-SyntaxHighlighter"
    >
      <span
        class="hljs-variable language_ css-g89m8z-Box"
        style="color: rgb(171, 86, 86);"
      >
        <span>
          console
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          .
        </span>
      </span>
      <span
        class="hljs-title function_ css-g89m8z-Box"
        style="color: rgb(136, 0, 0); font-weight: bold;"
      >
        <span>
          log
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          (
        </span>
      </span>
      <span
        class="hljs-string css-g89m8z-Box"
        style="color: rgb(136, 0, 0);"
      >
        <span>
          "Hello World"
        </span>
      </span>
      <span
        class="css-g89m8z-Box"
      >
        <span>
          )
        </span>
      </span>
    </code>
  </pre>
</div>
`;
